tests/reconfiguration.py

```diff
@@ -419,34 +428,66 @@ def test_learner_catches_up(network, args):
         assert len(rj["details"]["learners"]) == 0
 
         # At this point, there should be exactly one configuration, which includes the new node.
-        print(rj)
         assert len(rj["details"]["configs"]) == 1
         c0 = rj["details"]["configs"][0]["nodes"]
-        assert len(c0) == num_nodes_before + 1 and new_node.node_id in c0
+        assert len(c0) == num_nodes_before + 1
+        assert new_node.node_id in c0
 
     return network
 
 
-@reqs.description("Add a learner, suspend nodes, check that there is no progress")
+@reqs.description("Add a learner, partition nodes, check that there is no progress")
 def test_learner_does_not_take_part(network, args):
+    if network.partitioner is None:
+        return
+
     primary, backups = network.find_nodes()
-    nodes = network.get_joined_nodes()
-    f = infra.e2e_args.max_f(args, len(nodes)) + 1
-    f_backups = backups[:f]
+    f_backups = backups[: network.get_f() + 1]
 
     new_node = network.create_node("local://localhost")
     network.join_node(new_node, args.package, args, from_snapshot=False)
-    network.trust_node(new_node, args)
 
-    # No way to keep the new node suspended in learner state?
+    with network.partitioner.partition(f_backups):
```

Comment:
As discussed, can we move this to `partititons_test.py` as this requires `sudo`.

Reply:
Also, I don't understand why the partition of f+1 backups happens so early. This could be the reason why `new_node` cannot join the service (no commit over its join transaction). I believe it would be more pertinent to only isolate `new_node` from the rest of the service before `trust_node(new_node)` and then check that progress on the service doesn't depend on this node (e.g. by partitioning some other backups). Then, when all partitions are lifted, the node should join, observe its addition, orc and become part of consensus.

Reply:
I added the `CONFIGURATION` option in cmake scripts, so this one also runs with `sudo`.

Not sure I understand. `trust_node` runs right after the partition is introduced, I just added a `check_does_not_progress` in between for good measure. What would you have done differently?